src/Nix/Graph/Internal.hs

```diff
@@ -52,9 +53,12 @@ data Derivation = Derivation
   , derivationSystem :: Text
   , derivationInputDrvs :: [FilePath]
   , derivationBuilt :: Bool
+  , derivationEnv :: MapS.Map Text Text
   }
   deriving stock (Eq, Ord, Generic)
-  deriving anyclass (Hashable)
+
+instance Hashable Derivation where
+  hashWithSalt s = hashWithSalt s . derivationPath
```

Comment:
I think this ignores the `derivationBuilt` field, so we should stick with the derived instance.

Reply:
With the addition of a `Map`, the instance can no longer be derived.

Reply:
Will there be duplicates if the derivation is built/unbuilt.